ui/DesignSystem/Primitive/Input/Validated.svelte

```diff
@@ -0,0 +1,108 @@
+<script>
+  import Text from "../Text.svelte";
+  import Icon from "../Icon";
+  import Avatar from "../Avatar.svelte";
+
+  import { ValidationStatus } from "../../../src/validation.ts";
+
+  export let style = null;
+  export let placeholder = null;
+  export let value = null;
+  export let dataCy = null;
+
+  export let disabled = null;
+  export let variant = "vanilla"; // vanilla | avatar
```

Comment:
That this component has variants, where is avatar and imports the `Avatar` component is an indicator that it tries to too much and is too couple with a certain application of itself. Almost 50% of it's prop surface is to manipulate that part of the presentation.

A more reusable and less invasive way would be to allow [slots](https://svelte.dev/docs#slot) to be passed which are rendered in pre-defined spots. Similar to how the Remote component works: https://github.com/radicle-dev/radicle-upstream/blob/105a43081a27b3ad1f110df67f80a7befd0a7a89/ui/DesignSystem/Component/Remote.svelte#L7-L15

We can then build a higher-level component on top of the two primitives (Avatar and Validated), or combine them where needed, or use with ease in other contexts, without an avatar.

Reply:
We likely would get away with a single default slot for this use-case.

Reply:
Large part of my rationale is that the caller has the information on how to construct an Avatar already, as they are required by this component. Therefore the caller can construct the Avatar and pass it as a whole if desired.